Minor typos and Minor code refactor of fracture and stm_fracture_eat_a_chunk functions

nonyx777 opened the pull request:
stm_fracture_eat_a_chunk function processes all the chunk positions and returns an array of chunks, instead of processing them one by one.

nonyx777 committed:
minor typos

nonyx777 committed:
made stm_fracture_eat_a_chunk to process an array of positions, and return an array of meshes

nonyx777 committed:
shuffle happens outside the nested loop isolated. Happens as many times as it would inside the loops. Resulting in no conditionals inside the loops

cloudofoz commented:
Hi @nonyx777  thanks for sharing your modification, I really appreciate the work you put into it.

I’d like to explain why I haven’t merged the PR yet (still considering it), as there are a couple of points that are important for me in this context.

Regarding `stm_fracture_eat_a_chunk(...)`: when I originally wrote this function, I intentionally kept it as a single-operation function. The idea was to keep it flexible enough to be reused for other logic besides the current fracturing workflow, if needed in the future.

About the shuffling logic: this commit slightly changes the behavior compared to the original implementation. In my version, the ranges are reshuffled locally and consumed immediately (effectively using a freshly shuffled order at each nesting level). In the proposed change, the ranges are shuffled multiple times upfront and then iterated deterministically inside the nested loops.

Additionally, the number of shuffle calls is not the same:

Original logic:
- Y shuffled Z times
- X shuffled Z × Y times

Proposed logic:
- Y shuffled Z × Y times
- X shuffled Z × Y × X times (potentially a much higher number of shuffles)

That said, I do appreciate the effort and the intention behind the refactor, thanks again for your work!

Best,
C.

nonyx777 committed:
restored old shuffling code

nonyx777 commented:
Hey @cloudofoz, I agree with every point you made, and I actually thought about them.
I changed the shuffling code in the hopes that even though it will bring different results, it wouldn't affect the fracturing nature of the object and in the notion of why not precompute the data if i'm going to use it later. And you're right the number of shuffle codes is incorrect. However I have changed it back to what it was.
And about `stm_fracture_eat_a_chunk` I had a feeling it was intentional, but I thought it wouldn't be as destructive since you can still return an array with a single element.
